src/assign.cpp

```diff
@@ -913,15 +1013,39 @@ void fba::Assignment::assign_force(uint8_t tgtype, int32_t required_per_petal,
                 logger.warning(logmsg.str().c_str());
             }
         }
+        // Check final slitblock assignment counts
+        for (auto const & u : unique_slitblock) {
+            int32_t p = u.first;
+            int32_t s = u.second;
+            int32_t cur_slitblock = slitblock_count(tgtype, tile_id, p, s);
+            if (cur_slitblock < required_per_slitblock) {
+                logmsg.str("");
+                logmsg << "assign force " << tgstr << ": tile " << tile_id
+                       << ", petal " << p
+                       << ", slitblock " << s << " could only assign " << cur_slitblock
+                       << " (require " << required_per_slitblock
+                       << ").  Insufficient number of objects or too many collisions";
+                logger.warning(logmsg.str().c_str());
+            }
+        }
     }
 
     gtmname.str("");
     gtmname << "force " << tgstr << ": total";
     gtm.stop(gtmname.str());
 
     logmsg.str("");
-    logmsg << "Force assignment of " << required_per_petal << " "
-        << tgstr << " targets per petal";
+    if (required_per_petal && required_per_slitblock) {
```

Comment:
This logical check (and below) will return `true` even if the `required_per_*` values are -1.  Do you mean "`required_per_petal > 0`" and similar?

Reply:
Yes, thanks!!  fixed in 44e7cb9